application/src/main/java/org/togetherjava/tjbot/commands/help/OnGuildBanDeleteSpamThreadListener.java

```diff
@@ -0,0 +1,49 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.guild.GuildBanEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jooq.Result;
+
+import org.togetherjava.tjbot.commands.EventReceiver;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.HelpThreadsRecord;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.togetherjava.tjbot.db.generated.Tables.HELP_THREADS;
+
+/**
+ * Implements the functionality to delete recent help threads when the user gets banned.
+ */
+public final class OnGuildBanDeleteSpamThreadListener extends ListenerAdapter
+        implements EventReceiver {
+    private static final Duration recentThreadDuration = Duration.ofMinutes(30);
+
+    private final Database database;
+
+    /**
+     * @param database the database to get help threads from
```

Comment:
id be more specific on the motivation here `@param database to find help threads created by the banned user`

Reply:
done